proposals/utils/pdf_diff_logic.py

```diff
@@ -0,0 +1,1271 @@
+    )
+    row_fields = [
+        "get_metc_display",
+        "metc_details",
+        "metc_institution",
+        "get_is_medical_display",
+    ]
+
+    def get_row_fields(self):
+        rows = copy(self.row_fields)
+        obj = self.obj
+
+        if not obj.metc == "Y":
+            rows.remove("metc_details")
+            rows.remove("metc_institution")
+        else:
+            rows.remove("get_is_medical_display")
+
+        return rows
+
+
+class METCSection(PageBreakMixin, BaseSection):
+    """This class receives a proposal.wmo object."""
+
+    section_title = _("Aanmelding bij de METC")
+
+    row_fields = ["metc_application", "metc_decision", "metc_decision_pdf"]
+
+
+class TrajectoriesSection(PageBreakMixin, BaseSection):
+    """This class receives a proposal object."""
+
+    section_title = _("Eén of meerdere trajecten?")
+
+    row_fields = ["studies_similar", "studies_number"]
+
+    def get_row_fields(self):
+        rows = copy(self.row_fields)
+        obj = self.obj
+
+        if obj.studies_similar:
+            rows.remove("studies_number")
+
+        return rows
+
+
+class StudySection(BaseSection):
+    """This class receives a proposal.study object
+    Note the overwritten __init__ method for adding a sub_title."""
+
+    section_title = _("De deelnemers")
+    row_fields = [
+        "age_groups",
+        "legally_incapable",
+        "legally_incapable_details",
+        "has_special_details",
+        "special_details",
+        "traits",
+        "traits_details",
+        "necessity",
+        "necessity_reason",
+        "recruitment",
+        "recruitment_details",
+        "compensation",
+        "compensation_details",
+        "hierarchy",
+        "hierarchy_details",
+    ]
+
+    def __init__(self, obj):
+        super().__init__(obj)
+        self.sub_title = self.get_sub_title(self.obj, "study")
+
+    def get_row_fields(self):
+        rows = copy(self.row_fields)
+        obj = self.obj
+
+        if not has_adults(obj):
+            rows.remove("legally_incapable")
+            rows.remove("legally_incapable_details")
+        elif not obj.legally_incapable:
+            rows.remove("legally_incapable_details")
+        if not obj.has_special_details:
+            rows.remove("special_details")
+            rows.remove("traits")
+            rows.remove("traits_details")
+        elif not medical_traits(obj.special_details.all()):
+            rows.remove("traits")
+            rows.remove("traits_details")
+        elif not needs_details(obj.traits.all()):
+            rows.remove("traits_details")
+        if not necessity_required(obj):
+            rows.remove("necessity")
+            rows.remove("necessity_reason")
+        if not needs_details(obj.recruitment.all()):
+            rows.remove("recruitment_details")
+        if not obj.compensation or obj.compensation.needs_details:
+            rows.remove("compensation_details")
```

Comment:
I'm getting this field even tough I shouldn't. It seems that you forgot to invert the second part:

`if not obj.compensation or not obj.compensation.needs_details`